pkg/domain/infra/abi/images.go

```diff
@@ -170,7 +170,14 @@ func (ir *ImageEngine) Mount(ctx context.Context, nameOrIDs []string, opts entit
 	if opts.All {
 		listImagesOptions.Filters = []string{"readonly=false"}
 	}
-	images, err := ir.Libpod.LibimageRuntime().ListImages(ctx, nameOrIDs, listImagesOptions)
+
+	var images []*libimage.Image
+	var err error
+	if len(nameOrIDs) > 0 {
+		images, err = ir.Libpod.LibimageRuntime().ListImagesByNames(nameOrIDs)
+	} else {
+		images, err = ir.Libpod.LibimageRuntime().ListImages(ctx, listImagesOptions)
```

Comment:
I think it would be clearer to only set `listImagesOptions` inside the conditional.

… and then, probably, have the `if err != nil` check repeated inside each branch; that’s a bit tedious, but it would be more obvious that we didn’t forget.

---

It seems to me worth trying to combine all the code dealing with `opts.All`/`len(nameOrIDs)`:
- the initial “cannot mix” check
- this lookup
- determining `listMountsOnly`

in a single `switch`. It _might_ end up cleaner. I’m not 100% sure.

---

Whatever happens in this function, please update the `Unmount` code below the same way.